api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -200,6 +200,11 @@ type HostedControlPlaneSpec struct {
 	// +kubebuilder:validation:MaxProperties=20
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
+
+	// capabilities allows for disabling optional components at cluster install time.
+	// This field is optional and once set cannot be changed.
+	// +optional
+	Capabilities Capabilities `json:"capabilities,omitempty"`
```

Comment:
should we featuregate this instead of the inner field?
Let's also add soime validation tests https://github.com/openshift/hypershift/blob/607f618b03351a2313a24261b76e2f580f371ca3/test/e2e/create_cluster_test.go#L27

Reply:
We decided on slack that this can't be done at this point because the field is behind a tech preview feature gate, so we'll defer the tests for now.